### PR TITLE
Add missing chemical potential dependence to phase concentrations in GrandPotentialPFM.i

### LINKED ISSUE #16342: Provide missing chemical potential dependence to coupled_eta_function in GrandPotentialPFM.i
## Reason
This input file tests the implementation of the grand-potential phase-field model based on [[M. Plapp PRE 84, 031601 (2011)]](https://journals.aps.org/pre/pdf/10.1103/PhysRevE.84.031601). Whereas the phase concentrations `cs` and `cl` are defined as constants, according to Eq. 47 of the paper, they should be functions of the chemical potential `w`.

## Design
Chemical potential dependence can be provided by creating materials `cs` and `cl` using `DerivativeParsedMaterial`, which are then supplied to the `[coupled_eta_function]` block. The design will follow the input file for grand-potential model for multiphase, multicomponent system in [GrandPotentialMultiphase.i](https://github.com/idaholab/moose/blob/next/modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialMultiphase.i).

## Impact
Will give output consistent with analytic solution, even for starting conditions far from chemical equilibrium.

### PR DESCRIPTION
**Pull request by sourabhkadambi**
- `cs` and `cl` are now defined as functions of `w` using `DerivativeParsedMaterial` instead of constants
- these materials are supplied to `[coupled_eta_function]`
- `cseq` and `cleq` are newly defined as constants for equilibrium phase concentrations
- the fix will provide more accurate solutions for ICs far from chemical equilibrium
- improves the test case implementation of grand-potential model
closes #16342

### CODE REVIEW

**Comment by laagesen** [comment]
@loganharbour @dschwen I've been working with Sourabh on this so I can do the review if you want to assign me

File: modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i [hunk 1]
```diff
@@ -106,6 +106,24 @@
...
     eta = eta
   [../]
+  [./cs]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = cs
+    material_property_names = 'A cseq'
+    function = 'w/A + cseq' # since w = A*(c-cseq)
+    derivative_order = 2
+    enable_jit = false
```
Comment (laagesen) [inline modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i:116]: I would remove this, unless you have a reason to want to disable just-in-time compilation. The default value is true. Enabling just-in-time compilation will generally improve convergence at a very low penalty in start up time. Let me know if you want to discuss this system further.

File: modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i [hunk 2]
```diff
@@ -106,6 +106,24 @@
...
+  [../]
+  [./cl]
+    type = DerivativeParsedMaterial
+    args = 'w'
+    f_name = cl
+    material_property_names = 'A cleq'
+    function = 'w/A + cleq' # since w = A*(c-cleq)
+    derivative_order = 2
+    enable_jit = false
```
Comment (laagesen) [inline modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i:125]: suggest to remove this as above

**Review by laagesen — CHANGES_REQUESTED** [review]

**Comment by laagesen** [comment]
@loganharbour most of the tests are not running and currently have the message `Developer needed to activate` - any idea how to start the tests running again? Is anything needed from our end?

File: modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i (hunk 1 above)
Comment (sourabhkadambi) [reply modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i]: removed in next commit [c1f81df](https://github.com/idaholab/moose/pull/16343/commits/c1f81df6db659c9fc4367a80036eaab5f84514d5)

File: modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i (hunk 2 above)
Comment (sourabhkadambi) [reply modules/phase_field/test/tests/GrandPotentialPFM/GrandPotentialPFM.i]: removed in next commit [c1f81df](https://github.com/idaholab/moose/pull/16343/commits/c1f81df6db659c9fc4367a80036eaab5f84514d5)

**Review by laagesen — APPROVED** [review]

**Review by loganharbour — APPROVED** [review]

**Comment by loganharbour** [comment]
I'll merge this once we get next passing again.

**Pull request merged**